MapboxDirections/Match/MBMatchOptions.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+@objc(MBMatchOptions)
+open class MatchOptions: RouteOptions {
+    public init(tracePoints: [TracePoint], profileIdentifier: MBDirectionsProfileIdentifier?, resample: Bool?, timestamps: [Date]?) {
+        super.init(waypoints: tracePoints, profileIdentifier: profileIdentifier)
+    }
+    
+    @objc open var resample: Bool = false
+    
+    @objc open var timestamps: [Date]?
```

Comment:
Maintaining parallel arrays is prone to mismatches. Add a `Date`-typed `timestamp` property to Tracepoint or have the initializer take an array of `CLLocation`s (which has a `timestamp` property). Also consider adding a convenience initializer that takes an array of `CLLocationCoordinate2D`s, for when the developer doesn’t have timestamps to provide.

An open question is what to do if some tracepoints have timestamps and others do not. That can be a runtime check when computing a request URL.